src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -24,11 +29,57 @@ class SmartOrderAccepterBot(RandomProposerBot):
     def __init__(self, power_name, game) -> None:
         super().__init__(power_name, game)
         self.alliance_props_sent = False
+        self.stance = ScoreBasedStance(power_name, game)
+
+    def get_proposals(self, rcvd_messages: List[Tuple[int, Message]]) -> Dict[str, str]:
+        """
+        Extract proposal messages from received messages and checks for valid syntax before returning it
+        """
+        grammar = create_daide_grammar(level=130)
+        
+        # Extract messages containing PRP string
+        order_msgs = [msg[1] for msg in rcvd_messages if "PRP" in msg[1].message]
+
+        proposals = {}
+        for order_msg in order_msgs:
+            try:
+                # Parse using DAIDEPP functions
+                parse_tree = grammar.parse(order_msg.message)
+                output = daide_visitor.visit(parse_tree)
+
+                if output[0] == 'PRP':
+                    proposals[order_msg.sender] = order_msg.message
+            except Exception as e:
+                print(e)
+                pass
+
+        return proposals
+
+    def gen_pos_stance_messages(self, msgs_data: MessagesData) -> None:
+        """
+        Add messages to be sent to powers with positive stance. 
+        These messages would contain factual information about the orders that current power would execute in current round
+        """
+        orders_decided = FCT(ORR(XDO(self.orders.get_list_of_orders())))
+        for pow in self.stance.stance[self.power_name]:
+            if self.stance.stance[self.power_name][pow] > 0:
+                msgs_data.add_message(pow, str(orders_decided))
 
     def gen_messages(self, rcvd_messages):
```

Comment:
Can you modify the signature so that it takes in the order it is going to execute.

it should be the following (add a type hint): `def gen_messages(self, orders):`

Then generate the messages accordingly.

Reply:
And take the stance computation out of gen_messages then?